### DISCUSSION TITLE
Compare (diff) different versions of the same Helm chart

### DISCUSSION THREAD

**Discussion by alen-z** [artifacthub/hub#1872] [Ideas]
Hi,

I love Artifact Hub.

From my experience, I think community would value the ability to compare/diff different versions of the same Helm chart.

That would be awesome for few reasons:
* Check quickly for changes during upgrade. Changelog often does not exist or does not offer required level of detail to build confidence prior to the upgrade.
* Kudos to product teams from source Helm repositories that tag Helm charts properly, but it's often very hard to see diff between different versions because they are not tagged. One option is to download Helm charts locally and then compare —not very efficient.

What do you think?

Kind regards,
Alen

**Comment by tegioz** [comment]
Hi @alen-z 👋

This sounds like nice addition! We'll think a bit how we could make this work 🙂 

Thanks!

**Comment by tegioz** [comment]
Hope this works 😉 

<img width="1516" alt="templates-diff" src="https://user-images.githubusercontent.com/1213902/165129048-dfecd73b-cd3e-42fd-aa1a-e674a8f5cad7.png">

**Reply by alen-z** [reply]
Hey, that's awesome! You are quick to deliver — proper respect.

Only thing that seems very beneficial would be comparing `values.yaml` between different versions and then drill down to templates if needed (that you've made possible).

Nice work.

Edit: Another suggestion is to enable [sponsorship](https://github.com/sponsors) for this project.

**Reply by tegioz** [reply]
Thanks, we are glad you like it!

We'll do something similar for values soon, we'll keep you posted 😉

**Reply by tegioz** [reply]
<img width="1484" alt="compare-default-values" src="https://user-images.githubusercontent.com/1213902/171113652-a05b017c-d720-4dfc-ac6e-36669562bcd4.png">

**Reply by alen-z** [reply]
Such an awesome feature! Saves so much time for initial evaluation of changes introduced compared to previous version.

That's it from me :) Keep up excellent work you are doing.
